src/components/CurrentRefinements.vue

```diff
@@ -66,41 +79,21 @@ export default {
   },
   computed: {
     noRefinement() {
-      return this.refinements.length === 0;
-    },
-    refinements() {
-      let refinements = this.state.refinements.map(item => {
-        const attribute = item.type === 'query' ? 'query' : item.attributeName;
-        return {
-          type: item.type,
-          attribute,
-          label: item.computedLabel,
-          value: item,
-          key: `${attribute}-${item.computedLabel}`,
-        };
-      });
-
-      if (this.includedAttributes) {
-        refinements = refinements.filter(
-          ({ attribute }) => this.includedAttributes.indexOf(attribute) !== -1
-        );
-      } else {
-        refinements = refinements.filter(
-          ({ attribute }) => this.excludedAttributes.indexOf(attribute) === -1
-        );
-      }
-      return this.transformItems(refinements);
+      return this.state && this.state.items.length === 0;
     },
     widgetParams() {
       return {
-        clearsQuery:
-          this.excludedAttributes.indexOf('query') === -1 &&
-          (this.includedAttributes
-            ? this.includedAttributes.indexOf('query') !== -1
-            : true),
+        includedAttributes: this.includedAttributes,
+        excludedAttributes: this.excludedAttributes,
+        transformItems: this.transformItems,
       };
     },
   },
+  methods: {
+    createItemKey({ attribute, value, type, operator }) {
+      return [attribute, type, value, operator].join(':');
```

Comment:
Ah, that's why you added the boolean filter. However if you join `undefined` it will simply be coerced to an empty string, and thus it won't have an impact wether the string is unique within the list or not